test/expectations/code_action_resolve/nested_block_calls.exp.json

```diff
@@ -0,0 +1,47 @@
+{
+    "params": {
+        "kind": "refactor.rewrite",
+        "title": "Refactor: Switch block style",
+        "data": {
+            "range": {
+                "start": {
+                    "line": 0,
+                    "character": 0
+                },
+                "end": {
+                    "line": 3,
+                    "character": 3
+                }
+            },
+            "uri": "file:///fake"
+        }
+    },
+    "result": {
+        "title": "Refactor: Switch block style",
+        "edit": {
+            "documentChanges": [
+                {
+                    "textDocument": {
+                        "uri": "file:///fake",
+                        "version": null
+                    },
+                    "edits": [
+                        {
+                            "range": {
+                                "start": {
+                                    "line": 0,
+                                    "character": 29
+                                },
+                                "end": {
+                                    "line": 3,
+                                    "character": 3
+                                }
+                            },
+                            "newText": "{ |a| nested_call(fourth_call).each do |b|;  end }"
```

Comment:
That's a good point. I think we should be opinionated here, but I wasn't sure which way to go.

Essentially, we have two options:

- Find the first block within the selection and switch the style of that one. The implementation is straightforward and (at least on my opinion) the behaviour matches what I expect
- Find all blocks inside the selection and switch the style of all of them. This feels a bit surprising to me, but maybe others feel differently. Also, the implementation would need to be recursive since we don't know how many blocks are inside

Thoughts? cc @andyw8

Reply:
> Find the first block within the selection and switch the style of that one. The implementation is straightforward and (at least on my opinion) the behaviour matches what I expect

Do you mean we'd get:

```ruby 
method_call(other_call).each { |a| 
  nested_call(fourth_call).each do |b|
  end 
}
```

To me this is ok too if we want to be a bit more conservative. I simply do not expect to see `do; end` block being compressed to one-line.

But if I wanted to make the `do; end` block one line, I probably want to apply it to the second one too? In those cases, not having to it twice should be a lot more convenient.

Reply:
> do we base it on the first block we find and then turn all of them to that style?

Personally, this makes the most sense.